flink-runtime/src/test/java/org/apache/flink/runtime/leaderelection/LeaderChangeClusterComponentsTest.java

```diff
@@ -141,6 +142,16 @@ public void testReelectionOfJobMaster() throws Exception {
 		assertThat(jobResult.isSuccess(), is(true));
 	}
 
+	@Test
+	public void testTaskManagerRegisterReelectionOfResourceManager() throws Exception {
+
+		assertTrue(miniCluster.requestTaskManagerInfo().get().size() == NUM_TMS);
+		highAvailabilityServices.revokeResourceManagerLeadership().get();
+		highAvailabilityServices.grantResourceManagerLeadership().get();
+		Thread.sleep(500);
```

Comment:
We should try to use sleeps. I would suggest to use a polling loop with a lower sleep interval to make the test a bit more responsive.